astropy/visualization/log_linear_rgb.py

```diff
@@ -0,0 +1,380 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Combine 3 images to produce a properly-scaled RGB images with either
+log or linear scaling.
+
+Complementary to (and code structure inspired by)
+astropy.visualization.make_lupton_rgb.
+
+The three images must be aligned and have the same pixel scale and size.
+"""
+
+import numpy as np
+
+from astropy.visualization import (
+    LinearStretch,
+    LogStretch,
+    ManualInterval,
+)
+
+_OUTPUT_IMAGE_FORMATS = [float, np.float64, np.uint8]
+
+__all__ = ["make_rgb", "make_log_rgb", "make_linear_rgb"]
+
+
+class RGBImageMapping:
+    """
+    Class to map red, blue, green images into either a normalized float or
+    an 8-bit image, by performing optional clipping and applying
+    a scaling function to each band independently.
+
+    Parameters
+    ----------
+    minimum : float or array-like, optional
+        Intensity that should be mapped to black (a scalar or
+        array of R, G, B).
+    maximum : float or array-like, optional
+        Intensity that should be mapped to white (a scalar
+        or array of R, G, B).
+    stretch : `~astropy.visualization.BaseStretch` subclass instance
+        The stretch object to apply to the data. Default is
+        `~astropy.visualization.LinearStretch`.
+
+    """
+
+    def __init__(self, minimum=None, maximum=None, stretch=LinearStretch):
+        try:
+            len(minimum)
+        except TypeError:
+            minimum = 3 * [minimum]
+        if len(minimum) != 3:
+            raise ValueError("please provide 1 or 3 values for minimum.")
+
+        try:
+            len(maximum)
+        except TypeError:
+            maximum = 3 * [maximum]
+        if len(maximum) != 3:
+            raise ValueError("please provide 1 or 3 values for maximum.")
+
+        intervals = []
+        for i in range(3):
+            intervals.append(ManualInterval(vmin=minimum[i], vmax=maximum[i]))
+
+        self.intervals = intervals
+        self.stretch = stretch
+
+    def make_rgb_image(self, image_r, image_g, image_b, output_image_format=np.uint8):
+        """
+        Convert 3 arrays, image_r, image_g, and image_b into a RGB image,
+        either as an 8-bit per-channel or normalized image.
+
+        The input images can be int or float, and in any range or bit-depth,
+        but must have the same shape (NxM).
+
+        Parameters
+        ----------
+        image_r : ndarray
+            Image to map to red.
+        image_g : ndarray
+            Image to map to green.
+        image_b : ndarray
+            Image to map to blue.
+        output_image_format : numpy scalar type, optional
```

Comment:
Here and everywhere else this variable name is used: I think ``output_format_image`` should just be called ``output_dtype`` - format suggests e.g. JPG, PNG and so on (especially when it is available in the same function as one with ``filename=``. The ``image`` also seems unecessary.

Reply:
Done!